app/models/folio/item.rb

```diff
@@ -0,0 +1,29 @@
+# frozen_string_literal: true
+
+module Folio
+  # Represents an item returned from the /inventory-hierarchy/items-and-holdings Folio API
+  # TODO: This want's a "type" attribute, but I don't know how we get the folio version of a holding type.
+  #       See https://github.com/sul-dlss/searchworks_traject_indexer/blob/02192452815de3861dcfafb289e1be8e575cb000/lib/traject/config/sirsi_config.rb#L2379
+  Item = Data.define(:barcode, :status, :type, :callnumber, :public_note, :permanent_location, :temporary_location) do
+    def with_status(request_status)
+      ItemWithStatus.new(barcode:,
+                         status:,
+                         type:,
+                         callnumber:,
+                         public_note:,
+                         permanent_location:,
+                         temporary_location:,
+                         request_status:)
+    end
+
+    def self.from_hash(dyn)
+      new(barcode: dyn.fetch('barcode'),
```

Comment:
I feel like we've found examples of items that don't necessarily have all these fields. Is there a reason to be so strict?

Reply:
I don't like handling `nil` if there is no reason to. That's just added complexity for no reason.  Can you show examples where the API does not have these keys? I don't think I've seen any yet.

Reply:
https://github.com/sul-dlss/SearchWorks/issues/3198 fixed an issue with some similar looking code. Maybe the APIs do a better job keeping the data consistent, but I kinda doubt it.

Reply:
```
select count(*) from sul_mod_inventory_storage.item where lower(jsonb ->> 'barcode'::text) is null;
 count
-------
   173
```

Grrrr! Outliers.

Reply:
Welcome to libraries.

Reply:
https://folio-test.stanford.edu/inventory/view/a77ee8aa-ecc2-410d-a37a-69419677aeac/fef99dba-15f4-495c-b391-e501afdaed6d/a9030d19-5bbe-4976-ba31-ef8840f4145d